server/routes/prisonerList/pagedPrisonerSearchSummary.ts

```diff
@@ -97,11 +97,19 @@ export default class PagedPrisonerSearchSummary {
     (filterBy: FilterBy, value: string) =>
     (prisonerSearchSummary: PrisonerSearchSummary): boolean => {
       switch (filterBy) {
-        case FilterBy.NAME:
-          return (
-            prisonerSearchSummary.lastName.toUpperCase().includes(value.trim().toUpperCase()) ||
-            prisonerSearchSummary.firstName.toUpperCase().includes(value.trim().toUpperCase())
-          )
+        case FilterBy.NAME: {
+          const searchTerms = [
```

Comment:
This is essentially how the CIAG UI codebase does it, but I would argue ours is more readable, in a better class/file in the source tree, is testable .... and is actually tested! 😁 

For reference the CIAG version of this is buried in utils file 😬 
```
export function filterCiagList(
  profiles: PrisonerViewModel[],
  searchTerm: string,
  statusFilter: string,
): PrisonerViewModel[] {
  // Find matching profiles
  const filteredCiagList = profiles.map(p => ({
    ...p,
    searchTerms: [
      p.prisonerNumber.toLowerCase(),
      p.firstName.toLowerCase(),
      p.lastName.toLowerCase(),
      `${p.firstName.toLowerCase()}, ${p.lastName.toLowerCase()}`,
      `${p.firstName.toLowerCase()} ${p.lastName.toLowerCase()}`,
      `${p.lastName.toLowerCase()}, ${p.firstName.toLowerCase()}`,
      `${p.lastName.toLowerCase()} ${p.firstName.toLowerCase()}`,
      `${p.firstName.charAt(0).toLowerCase()} ${p.lastName.toLowerCase()}`,
    ].join('|'),
  }))

  const filteredSearch: () => PrisonerViewModel[] = () => {
    let filteredList = filteredCiagList
    if (searchTerm) {
      filteredList = filteredList.filter(x =>
        x.searchTerms.split('|').some(term => term.includes(searchTerm.toLowerCase())),
      )
    }

    if (statusFilter) {
      filteredList = filteredList.filter(x => x.status === statusFilter)
    }

    return filteredList
  }

  return filteredSearch()
}
```